pkg/container/kubernetes/client.go

```diff
@@ -1043,68 +1041,57 @@ func configureContainer(
+		if container.SecurityContext.RunAsNonRoot == nil && platformContext.RunAsNonRoot != nil {
+			container.SecurityContext = container.SecurityContext.WithRunAsNonRoot(*platformContext.RunAsNonRoot)
 		}
 
-		if container.SecurityContext.Privileged == nil {
-			container.SecurityContext = container.SecurityContext.WithPrivileged(false)
+		if container.SecurityContext.RunAsUser == nil && platformContext.RunAsUser != nil {
+			container.SecurityContext = container.SecurityContext.WithRunAsUser(*platformContext.RunAsUser)
 		}
 
-		if container.SecurityContext.ReadOnlyRootFilesystem == nil {
-			container.SecurityContext = container.SecurityContext.WithReadOnlyRootFilesystem(true)
+		if container.SecurityContext.RunAsGroup == nil && platformContext.RunAsGroup != nil {
+			container.SecurityContext = container.SecurityContext.WithRunAsGroup(*platformContext.RunAsGroup)
 		}
 
-		if container.SecurityContext.AllowPrivilegeEscalation == nil {
-			container.SecurityContext = container.SecurityContext.WithAllowPrivilegeEscalation(false)
+		if container.SecurityContext.AllowPrivilegeEscalation == nil && platformContext.AllowPrivilegeEscalation != nil {
+			container.SecurityContext = container.SecurityContext.WithAllowPrivilegeEscalation(*platformContext.AllowPrivilegeEscalation)
 		}
-	}
 
-	if platform == PlatformOpenShift {
-		logger.Infof("Setting OpenShift security context requirements to container %s", *container.Name)
-
-		if container.SecurityContext.RunAsUser != nil {
-			container.SecurityContext.RunAsUser = nil
+		if container.SecurityContext.ReadOnlyRootFilesystem == nil && platformContext.ReadOnlyRootFilesystem != nil {
+			container.SecurityContext = container.SecurityContext.WithReadOnlyRootFilesystem(*platformContext.ReadOnlyRootFilesystem)
 		}
 
-		if container.SecurityContext.RunAsGroup != nil {
-			container.SecurityContext.RunAsGroup = nil
+		if container.SecurityContext.SeccompProfile == nil && platformContext.SeccompProfile != nil {
+			container.SecurityContext = container.SecurityContext.WithSeccompProfile(platformContext.SeccompProfile)
 		}
 
-		if container.SecurityContext.SeccompProfile == nil {
-			container.SecurityContext.SeccompProfile =
-				corev1apply.SeccompProfile().WithType(
-					corev1.SeccompProfileTypeRuntimeDefault)
-		} else {
-			container.SecurityContext.SeccompProfile =
-				container.SecurityContext.SeccompProfile.WithType(
-					corev1.SeccompProfileTypeRuntimeDefault)
+		if container.SecurityContext.Capabilities == nil && platformContext.Capabilities != nil {
+			container.SecurityContext = container.SecurityContext.WithCapabilities(platformContext.Capabilities)
 		}
 
-		if container.SecurityContext.Capabilities == nil {
-			container.SecurityContext.Capabilities = &corev1apply.CapabilitiesApplyConfiguration{
-				Drop: []corev1.Capability{"ALL"},
+		// For OpenShift, override certain fields even if they exist
+		if platform == PlatformOpenShift {
```

Comment:
I wonder if we could have an if-else or switch for the platform that would skip setting the fields that OCP nils anyway. But that might be just personal preference and doesn't affect functionality

Reply:
Perhaps the differentiation in the value per approach will come when there are more platforms added that require specific configuration?

Reply:
sounds good